src/internal/packager2/load.go

```diff
@@ -0,0 +1,212 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+package packager2
+
+import (
+	"archive/tar"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/defenseunicorns/pkg/helpers/v2"
+	"github.com/mholt/archiver/v3"
+
+	"github.com/zarf-dev/zarf/src/config"
+	"github.com/zarf-dev/zarf/src/pkg/layout"
+	"github.com/zarf-dev/zarf/src/pkg/packager/filters"
+	"github.com/zarf-dev/zarf/src/pkg/packager/sources"
+	"github.com/zarf-dev/zarf/src/pkg/utils"
+	"github.com/zarf-dev/zarf/src/types"
+)
+
+// LoadOptions are the options for LoadPackage.
+type LoadOptions struct {
+	Source                  string
+	Shasum                  string
+	PublicKeyPath           string
+	SkipSignatureValidation bool
+	Filter                  filters.ComponentFilterStrategy
+}
+
+// LoadPackage optionally fetches and loads the package from the given source.
+func LoadPackage(ctx context.Context, opt LoadOptions) (*layout.PackagePaths, error) {
+	srcType, err := identifySource(opt.Source)
+	if err != nil {
+		return nil, err
+	}
+
+	tarDir, err := utils.MakeTempDir(config.CommonOptions.TempDirectory)
+	if err != nil {
+		return nil, err
+	}
+	defer os.RemoveAll(tarDir)
+	tarPath := filepath.Join(tarDir, "packager.tar")
```

Comment:
Should the `tar` specific stuff here be moved into the tar case statement?

Reply:
No because each case will in the end produce a tar of the package. I do think it would be good to make it its own function though.